### ISSUE TITLE
Consider exporting functions as `default`

### ISSUE THREAD

**Issue by tomekwi** [jussi-kalliokoski/trine#56]
My two cents to #46:

When I started using *trine* it was a surprising discovery that trying to `import map from 'trine/iterable/map'` failed. It cost me some time to find out that I should have `import`ed `{map}` instead.

Since I heard that *[“Default exports are favored”](http://www.2ality.com/2014/09/es6-modules-final.html#default_exports_are_favored)* I’ve always tried to only `export default` out of my modules, and I’ve seen others do the same. I’ve grown to expect that of other ES 2015+ libraries as well.

Is there any reason why you use named exports instead of `export`ing `default`? That would make for [even simpler syntax](https://github.com/jussi-kalliokoski/trine/pull/46#issuecomment-125577935):

```js
import dropTail from "trine/iterable/droptail";
import dropHead from "trine/iterable/dropHead";
import intersection from "trine/iterable/intersection";
import sortAlphabetically from "trine/iterable/sortAlphabetically";
import uniq from "trine/iterable/uniq";
import difference from "trine/iterable/difference";
import enumerate from "trine/iterable/enumerate";
import zip from "trine/iterable/zip";
import toFunction from "trine/value/toFunction";
import partial from "trine/function/partial";
```

**Comment by jussi-kalliokoski** [comment]
> Is there any reason why you use named exports

Yes, actually. This allows code editors to know the name of the export, so for example one could imagine having a keyboard shortcut for importing a new thing and just type in the module path (or part of it) and the editor can handle the rest, including naming. :)

Also some of the modules have multiple exports:

```javascript
import { partial, _, ___ } from "trine/function/partial";
```

**Comment by tomekwi** [comment]
>  This allows code editors to know the name of the export

Sincerely, I don’t think basing a design decision on features of a code editor is the way to go.

When you `import map from 'trine/iterable/map'`, it’s just as easy to determine the name of the export – it’s `default`. A good editor should figure that out.

>  one could imagine having a keyboard shortcut for importing a new thing and just type in the module path (or part of it) and the editor can handle the rest, including naming. :)

On Sublime I used to have a snippet for that: https://github.com/tomekwi/sublime-snippets/blob/ae89045999dff04977174fc27a694df17081364c/JS/import.sublime-snippet – no need for named exports there.

> Also some of the modules have multiple exports:

No problem:

```js
import { default as partial, _, ___ } from "trine/function/partial";
```

**Comment by edge** [comment]
It's actually even easier than that.

```js
import partial, { _, ___ } from 'trine/function/partial';
```

**Comment by tomekwi** [comment]
Wow, good to know! :)

**Comment by jussi-kalliokoski** [comment]
Hmm, I'm a bit torn in between on this - I have an ideal nagging in my head that everything should be named, but then again default exports are the most convenient way to use libraries like this that export (mostly) one thing per module. :D I'd love to get more input, but I'm pretty sure you're right about this being the right way forward.

**Comment by sladiri** [comment]
If editors need named exports and can even provide some GUI to make imports easier, I can live with the extra brackets just fine. So I am a bit surprised about your last comment @jussi-kalliokoski. :)
